From 31137f8527db8c497a36585a2b76bda23a6410f2 Mon Sep 17 00:00:00 2001
From: flar2 <asegaert@gmail.com>
Date: Fri, 15 Nov 2013 23:18:01 -0500
Subject: [PATCH 426/507] mdss_dsi: do not power down panel when sweep2wake is
 enabled In order to detect touches during suspend,
 the panel must be powered on. When sweep2wake is
 enabled, change one of the panel-off-cmds in
 mdss_dsi_panel.c from 0x10 to 0x11.

Signed-off-by: flar2 <asegaert@gmail.com>
Signed-off-by: Simarpreet Singh <simar@linux.com>
---
 drivers/video/msm/mdss/mdss_dsi_panel.c |   12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/drivers/video/msm/mdss/mdss_dsi_panel.c b/drivers/video/msm/mdss/mdss_dsi_panel.c
index 5f80d3c..0990d84 100644
--- a/drivers/video/msm/mdss/mdss_dsi_panel.c
+++ b/drivers/video/msm/mdss/mdss_dsi_panel.c
@@ -25,6 +25,9 @@
 #include <linux/debugfs.h>
 #include <linux/ctype.h>
 #endif
+#ifdef CONFIG_TOUCHSCREEN_SWEEP2WAKE
+#include <linux/input/sweep2wake.h>
+#endif
 
 #include "mdss_dsi.h"
 
@@ -315,6 +318,15 @@ static int mdss_dsi_panel_off(struct mdss_panel_data *pdata)
 	if (!gpio_get_value(ctrl->disp_en_gpio))
 		return 0;
 
+#ifdef CONFIG_TOUCHSCREEN_SWEEP2WAKE
+	if (s2w_switch) {
+		ctrl->off_cmds.cmds[1].payload[0] = 0x11;
+	} else {
+		ctrl->off_cmds.cmds[1].payload[0] = 0x10;
+	}
+	pr_info("[sweep2wake] payload = %x \n", ctrl->off_cmds.cmds[1].payload[0]);
+#endif
+
 	if (ctrl->off_cmds.cmd_cnt)
 		mdss_dsi_panel_cmds_send(ctrl, &ctrl->off_cmds);
 
-- 
1.7.9.5

